### PR TITLE
Introduce helper function name policy

### PR DESCRIPTION
**Pull request by tcharding**
As much as it hurts the C hacker inside me we have settled on using `_internal` to mark private function names that clash with a public function of the same name.

Introduce a policy section and rename one instance, I did not grep the codebase looking for other violations.

This came up because I had to look at what `_inner` implied when reading the function name somewhere else.

### CODE REVIEW

**Comment by shinghim** [comment]
Just for some context, is there a discussion that you can link for where we settled on using `_internal`, or is it more that there is existing code that uses `_internal` already?

**Comment by tcharding** [comment]
I don't have links sorry, its from multiple PRs where I tried to use the terse `_foo()` (as often seen in kernel code). And @Kixunil requested changes to not do so because he correctly asserted that in Rust a leading `_` has a meaning already (tells the linter that variable is unused).

**Review by shinghim — APPROVED** [review]
ACK 08f1d3bc55ed52c953b309d31f13c507cda7b47a

**Review by Kixunil — APPROVED** [review]
ACK 08f1d3bc55ed52c953b309d31f13c507cda7b47a

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -175,6 +175,7 @@ in Bitcoin Core, with the following exceptions:
 - the case should follow Rust standards (i.e. PascalCase for types and
   snake_case for fields and variants);
 - omit `C`-prefixes.
+- If Function `foo` needs an private helper function use `foo_internal`.
```
Comment (storopoli) [inline CONTRIBUTING.md:178]: Also why titlecasing `Function`?
```suggestion
- If function `foo` needs a private helper function, use `foo_internal`.
```

[the remark this answers is not in the document]
Comment (tcharding) [reply CONTRIBUTING.md:178]: Oh yes, thanks for reviewing! Will fix.

Comment (tcharding) [reply CONTRIBUTING.md:178]: Ugh, sloppy PR by me. Thanks man, will use as suggested.

**Comment by tcharding** [comment]
I added an additional patch to clean up the three list items also (based on our discussion a few weeks ago @storopoli).

**Review by storopoli — APPROVED** [review]
ACK 84df3438caa58263a9cbc8a41eace9c1f0fdca6b

**Review by shinghim — APPROVED** [review]
ACK 0a45c68cf8625b211109e2dbb528c73b9fa00caf

**Review by apoelstra — APPROVED** [review]
ACK 84df3438caa58263a9cbc8a41eace9c1f0fdca6b successfully ran local tests; sgtm. should probably update rust-miniscript which uses a `real_` prefix

**Comment by apoelstra** [comment]
Oh, oops, @Kixunil I probably should have waited for you on this. Because it only affected private functions and it was a minor thing I considered it a "trivial change" that I could one-ack merge but it *is* a policy change so I guess it's not trivial.

**Comment by apoelstra** [comment]
The diff between your most recent ACK and the commit that I merged consists of only grammar changes.

**Pull request merged**